_source/_posts/2020-12-11-build-and-secure-an-api-in-python-with-fastapi.md

```diff
@@ -0,0 +1,362 @@
+
+### Install FastAPI
+
+Start by creating a new Python project and using pip to [install FastAPI](https://fastapi.tiangolo.com/#installation) and [Uvicorn](https://www.uvicorn.org/) as your asynchronous server:
+
+```bash
+pip install fastapi uvicorn
+pip freeze > requirements.txt
+```
+
+The dependencies will be added to your requirements.txt file.
+
+### Create a New Endpoint
+
+For this tutorial, you'll use a single `main.py` file that contains all your routes. You can [break this file up as your application grows](https://fastapi.tiangolo.com/tutorial/bigger-applications/), but since you'll just be adding a couple of endpoints, you don't need to worry about that now.
+
+Create the `main.py` file in your project's root directory and add the following:
+
+```python
+from fastapi import FastAPI
+
+
+app = FastAPI()
+
+
+# Hello World route
+@app.get("/")
+def read_root():
+    return {"Hello": "World"}
+
+```
+
+### Accessing Your API
+
+When using FastAPI, there are two ways you can access your API. You can use an API explorer like [Postman](https://www.postman.com/), or you can use FastAPI's interactive documentation in your browser. This tutorial will focus on the latter.
+
+To start your application, run the Uvicorn server:
+
+```bash
+uvicorn main:app --reload
+```
+
+Then, open [http://127.0.0.1:8000](http://127.0.0.1:8000) in your browser to see your "Hello, world" endpoint in action:
+
+![A "Hello, World" endpoint in FastAPI](https://i.imgur.com/XHwSIuy.png)
+
+FastAPI's interactive documentation is available at [http://127.0.0.1:8000/docs](http://127.0.0.1:8000/docs). Once you're there, you'll see details about each endpoint automatically. FastAPI uses type hints and context in your application to produce these docs on its own.
+
+![FastAPI interactive documentation](https://i.imgur.com/Va63heZ.png)
+
+Now that you've seen how easy it is to get started, you're ready to build a more useful application. In the remainder of this tutorial, you'll see how to create protected endpoints in FastAPI that use Okta as an authorization server.
+
+## Securing FastAPI with Okta
+
+To demonstrate some of the more advanced features of the FastAPI framework, I'll show you how to build a protected endpoint that uses [the client credentials flow](https://developer.okta.com/docs/guides/implement-client-creds/overview/) to authorize access. By the end of this tutorial, you will be able to enter your Okta client ID and secret into FastAPI's interactive docs to get an access token. You'll see how to validate this JWT access token remotely using Okta's [introspect](https://developer.okta.com/docs/reference/api/oidc/#introspect) endpoint and locally using the [Okta JWT Python library](https://github.com/adithyasampatoor/okta_jwt).
```

Comment:
Links to d.o.c should be relative so they work in preview. Also, is there no official Okta Python library?